internal/restic/node_windows.go

```diff
@@ -351,32 +361,84 @@ func decryptFile(pathPointer *uint16) error {
+// Created time and Security Descriptors.
+// It also checks if the volume supports extended attributes and stores the result in a map
+// so that it does not have to be checked again for subsequent calls for paths in the same volume.
 func (node *Node) fillGenericAttributes(path string, fi os.FileInfo, stat *statT) (allowExtended bool, err error) {
 	if strings.Contains(filepath.Base(path), ":") {
-		//Do not process for Alternate Data Streams in Windows
+		// Do not process for Alternate Data Streams in Windows
 		// Also do not allow processing of extended attributes for ADS.
 		return false, nil
 	}
-	if !strings.HasSuffix(filepath.Clean(path), `\`) {
-		// Do not process file attributes and created time for windows directories like
-		// C:, D:
-		// Filepath.Clean(path) ends with '\' for Windows root drives only.
-		var sd *[]byte
-		if node.Type == "file" || node.Type == "dir" {
-			if sd, err = fs.GetSecurityDescriptor(path); err != nil {
-				return true, err
-			}
+
+	if strings.HasSuffix(filepath.Clean(path), `\`) {
+		// filepath.Clean(path) ends with '\' for Windows root volume paths only
+		// Do not process file attributes, created time and sd for windows root volume paths
+		// Security descriptors are not supported for root volume paths.
+		// Though file attributes and created time are supported for root volume paths,
+		// we ignore them and we do not want to replace them during every restore.
+		allowExtended, err = checkAndStoreEASupport(path)
+		if err != nil {
+			return false, err
+		}
+		return allowExtended, nil
+	}
+
+	var sd *[]byte
+	if node.Type == "file" || node.Type == "dir" {
+		// Check EA support and get security descriptor for file/dir only
+		allowExtended, err = checkAndStoreEASupport(path)
+		if err != nil {
+			return false, err
+		}
+		if sd, err = fs.GetSecurityDescriptor(path); err != nil {
+			return allowExtended, err
 		}
+	}
+	// Add Windows attributes
+	node.GenericAttributes, err = WindowsAttrsToGenericAttributes(WindowsAttributes{
+		CreationTime:       getCreationTime(fi, path),
+		FileAttributes:     &stat.FileAttributes,
+		SecurityDescriptor: sd,
+	})
+	return allowExtended, err
+}
+
+// checkAndStoreEASupport checks if the volume of the path supports extended attributes and stores the result in a map
+// If the result is already in the map, it returns the result from the map.
+func checkAndStoreEASupport(path string) (isEASupportedVolume bool, err error) {
+	// Check if it's an extended length path
+	if strings.HasPrefix(path, uncPathPrefix) {
+		// Convert \\?\UNC\ extended path to standard path to get the volume name correctly
+		path = `\\` + path[len(uncPathPrefix):]
+	} else if strings.HasPrefix(path, extendedPathPrefix) {
+		//Extended length path prefix needs to be trimmed to get the volume name correctly
+		path = path[len(extendedPathPrefix):]
+	} else if strings.HasPrefix(path, globalRootPrefix) {
+		// EAs are not supported for \\?\GLOBALROOT i.e. VSS snapshots
```

Comment:
I made a mistake here while fixing the test case errors. 
Extended attributes are in fact supported for VSS snapshots.
Instead of returning false, we should be handling the volume names for VSS snapshots correctly. Apologies, I can add a separate PR.

Reply:
> Instead of returning false, we should be handling the volume names for VSS snapshots correctly. Apologies, I can add a separate PR.

Does `GetVolumeInformation` support such paths or do we needs even more weird special case handling?

Reply:
We need to get the 'volumeName' path like this - 
        
```
// Extract the VSS snapshot volume name eg. `\\?\GLOBALROOT\Device\HarddiskVolumeShadowCopyXX`
if parts := strings.SplitN(path, `\`, 7); len(parts) >= 6 {
	volumeName = strings.Join(parts[:6], `\`)
}
```
		
Rest of the code remains same-

fs.PathSupportsExtendedAttributes(volumeName + `\`) i.e. GetVolumeInformation works.

Reply:
Sigh, I'm not happy about all those Windows special case. But looks like we'd better add them. Can you open a PR for it?

Reply:
Yeah, problem is filepath.VolumeName does not do any windows specific handling, it's just syntactical. 
We can probably get actual volumeName using windows.GetVolumeInformation itself, but it would mean we need to make the system call for every path. 
Right now, we are doing all that manual processing to avoid the system call.